File: 이상욱/BOJ/m2w7/BOJ14425.java
package m2w7;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.HashSet;
import java.util.StringTokenizer;

public class BOJ14425 {
	
	public static void main(String[] args) throws Exception{
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		
		int n = Integer.parseInt(st.nextToken());
		int m = Integer.parseInt(st.nextToken());
		
		int ans = 0;
		
		HashSet<String> hs = new HashSet<>();
		
		for(int i = 0; i < n; i++) {
			
			hs.add(br.readLine());
			
		}
		
		for(int i = 0; i < m; i++) {
			
			String str = br.readLine();
			
			if(hs.contains(str)) {
				ans++;
			}
			
		}
		
		System.out.println(ans);
		
	}
}

File: 이상욱/BOJ/m2w7/BOJ14425.kt
package m2w7

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*
import kotlin.collections.HashSet

fun main(){

    val br = BufferedReader(InputStreamReader(System.`in`))
    val st = StringTokenizer(br.readLine())

    val N = st.nextToken().toInt()
    val M = st.nextToken().toInt()
    var ans = 0
    val stringSet = HashSet<String>()

    repeat(N){
        stringSet.add(br.readLine())
    }

    repeat(M){
        val str = br.readLine()

        if(str in stringSet){
            ans++
        }
    }

    println(stringSet)

    println(ans)


}

